kawin/tests/test_diffusion.py

```diff
@@ -0,0 +1,114 @@
+
+def test_SinglePhaseFluxes():
+    '''
+    Tests the dimensions of the single phase fluxes function
+
+    Should be (E-1, N+1) where E is the number of elements and
+    N is the number of points
+    '''
+    singleModelBinary.reset()
+    singleModelBinary.setCompositionStep(0.2, 0.8, 0, 'CR')
+    singleModelBinary.setThermodynamics(NiCrTherm)
+    singleModelBinary.setTemperature(800)
+    singleModelBinary.setup()
+
+    fBinary, _ = singleModelBinary.getFluxes()
+
+    singleModelTernary.reset()
+    singleModelTernary.setCompositionStep(0.2, 0.4, 0, 'CR')
+    singleModelTernary.setCompositionStep(0.4, 0.4, 0, 'AL')
+    singleModelTernary.setThermodynamics(NiCrAlTherm)
+    singleModelTernary.setTemperature(800)
+    singleModelTernary.setup()
+
+    fTernary, _ = singleModelTernary.getFluxes()
+
+    assert(fBinary.shape == (1,N+1))
+    assert(fTernary.shape == (2,N+1))
+
+def test_HomogenizationMobility():
+    '''
+    Tests the dimensions of the homogenization mobility function
+
+    Should be (P, E, N) where P is the number of phases,
+    E is the number of elements and
+    N is the number of points
+    '''
+    homogenizationBinary.reset()
+    homogenizationBinary.setCompositionStep(0.2, 0.8, 0, 'CR')
+    homogenizationBinary.setThermodynamics(NiCrTherm)
+    homogenizationBinary.setTemperature(800)
+    homogenizationBinary.setup()
+
+    mobBinary = homogenizationBinary.getMobility(homogenizationBinary.x)
+
+    homogenizationTernary.reset()
+    homogenizationTernary.setCompositionStep(0.2, 0.4, 0, 'CR')
+    homogenizationTernary.setCompositionStep(0.4, 0.4, 0, 'AL')
+    homogenizationTernary.setThermodynamics(NiCrAlTherm)
+    homogenizationTernary.setTemperature(800)
+    homogenizationTernary.setup()
+
+    mobTernary = homogenizationTernary.getMobility(homogenizationTernary.x)
+
+    assert(mobBinary.shape == (len(homogenizationBinary.phases),2,N))
+    assert(mobTernary.shape == (len(homogenizationTernary.phases),3,N))
+
+def test_homogenizationSinglePhaseMobility():
+    '''
+    Tests that in a single phase region, any of the mobility functions will give 
+    the same mobility of the single phase itself
+    '''
+    homogenizationTernary.reset()
+    homogenizationTernary.setCompositionStep(0.1, 0.4, 0, 'CR')
+    homogenizationTernary.setCompositionStep(0.1, 0.4, 0, 'AL')
+    homogenizationTernary.setThermodynamics(NiCrAlTherm)
+    homogenizationTernary.setTemperature(800)
+    homogenizationTernary.setup()
+
+    mob = homogenizationTernary.getMobility(homogenizationTernary.x)
+
+    mobFuncs = ['wiener upper', 'wiener lower', 'hashin upper', 'hashin lower', 'lab']
+    mobs = []
+    for f in mobFuncs:
+        homogenizationTernary.clearCache()
+        homogenizationTernary.setup()
+        homogenizationTernary.setMobilityFunction(f)
+        mobs.append(homogenizationTernary.mobilityFunction(homogenizationTernary.x))
+        assert(np.allclose(mobs[-1][:,0], mob[0,:,0], rtol=1e-3))
+
```

Comment:
These tests are fine, but do you have any thoughts about how to verify numerical correctness of the values, such as compositions, phase fractions, fluxes, etc.? At least I think we should be checking that the values do not change over time, i.e., calculate a result today and use it as the check value for a test.

Reply:
I added tests for each mobility averaging function for whether the model is in a single phase or two phase region.